lib/policies/load-balancing.js

```diff
@@ -209,20 +209,22 @@ DCAwareRoundRobinPolicy.prototype.newQueryPlan = function (keyspace, queryOption
     this.index = 0;
   }
   this._sliceNodesByDc();
+  // Use a local reference of hosts
+  var localHostsArray = this.localHostsArray;
```

Comment:
It took me a little bit to wrap my head around this, is this what was happening previously to cause the bug?

1. user executes query, which calls newQueryPlan to resolve hosts.
2. a Host is added or removed before/during the query plan is iterated over.  As part of this _cleanHostCache is called which sets localHostsArray/remoteHostsArray to null.
3. query plan is iterated over, because localHostsArray is null, we hit the failure.

Getting a local reference of these arrays makes it so if `this.localHostsArray` is set to null, the previous array is still maintained by this local reference, so we can iterate over the host map made at the time of newQueryPlan being called.